### PR TITLE
Break apart georeferenceable interface

### PR DESCRIPTION
**Pull request by nithinp7**
Now we have a very streamlined `ICesiumGeoreferenceListener` which subscribes for notifications of changes in the world georeference. We also now have a `ICesiumBoundingVolumeProvider` that can contribute its bounding volume to origin placement calculations (when the origin placement mode is "bounding volume origin").

Fixes: https://github.com/CesiumGS/cesium-unreal/issues/321

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @nithinp7!

* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

File: Source/CesiumRuntime/Public/CesiumGeoreferenceListener.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ */
+class CESIUMRUNTIME_API ICesiumGeoreferenceListener {
+  GENERATED_BODY()
+
+public:
+  /**
+   * Notifies georeferenced objects that the underlying georeference transforms
+   * have been updated. The new transforms can be fetched from the
+   * `CesiumWorldGeoreference` actor.
```
Comment (javagl) [inline Source/CesiumRuntime/Public/CesiumGeoreferenceListener.h:25]: This *might* be `CesiumGeoreference`, but ... this wouldn't necessarily tell the user where to obtain the transforms. Some options for "tansporting" this information range from

1, You have to store a reference to the GeoRef, and fetch the info from that reference on your own
2. The method could receive the `GeoRef*` that was updated (i.e. the one that triggered that call)
3. The method could receive ~"something like the updated transform"

For 1., the receiver of this call would not know which GeoRef was updated. For 3. it's not clear *which* information that is (it's more than a 4x4 matrix, for sure). In other contexts, I'd pragmatically go for 2. (and consider something like a `GeoRefChangedEvent` instead, if more information would have to be transported through that call).

But for 2., the caller could once again *store* this reference that he receives there, raising further questions about its validity and such. Depending on the other decisions, the method could also receive "an identifier" for the GeoRef, like `NotifyGeoreferenceUpdated("THE_DEFAULT_ONE")` vs. `NotifyGeoreferenceUpdated("MY_SPECIAL_ONE")`, where this "identifier" is used to access some sort of global/singleton-like instance.

(tl;dr: This has to be sorted out, and aligned with the (coarse-grained) GeoRef refactoring....)

**Comment by xuelongmu** [comment]
I haven't yet had a chance to take a closer look at the code, but just wanted to bring up some of our discussions from #330, especially regarding the [delegates](https://github.com/CesiumGS/cesium-unreal/issues/330#issuecomment-820074706) for the `NotifyGeoreferenceUpdated` - do you think it would be a good idea to switch over to delegates instead of a C++ interface?

Some advantages would be being able to register a callback from blueprints instead of implementing the interface in a C++ class, and hiding the actual registration in a static function like what @javagl suggested. One stretch goal of this would be to handle multiple georeferences as well, through the static function library, using a string identifier perhaps.

**Review by baothientran — COMMENTED** [review]
This looks too me @nithinp7, I have some comments regarding about the listener interface

**Comment by kring** [comment]
I think we can close this because it is included in #459 and we're unlikely to merge it separately from the larger georeference refactoring. Please reopen if I'm mistaken.

**Pull request closed without merging**